androidApp/src/main/java/com/mbta/tid/mbta_app/android/pages/MapAndSheetPage.kt

```diff
@@ -335,13 +335,13 @@ fun MapAndSheetPage(
                 if (sheetNavEntrypoint != null) navController.navigate(sheetNavEntrypoint)
                 else navController.popBackStack(SheetRoutes.Entrypoint::class, inclusive = false)
             },
-            sheetBackState =
+            backButtonPresentation =
                 if (
                     nearbyTransit.scaffoldState.bottomSheetState.currentValue == SheetValue.Large ||
                         hideMaps
                 )
-                    NavigationCallbacks.SheetBackState.Shown
-                else NavigationCallbacks.SheetBackState.Hidden,
+                    NavigationCallbacks.BackButtonPresentation.Header
+                else NavigationCallbacks.BackButtonPresentation.Floating,
```

Comment:
If you check the current nav entry here, you can pass the navCallbacks directly to the HomeMapView, and then the map and the sheet will always agree about which of them owns the back button.